### ISSUE TITLE
Сделать в запросе к overpass api фильтрацию на основе значений тега highway.

### ISSUE THREAD

**Issue by GeorgiySurkov** [webebers/city-view#35]

**Comment by GeorgiySurkov** [comment]
В ходе выполнения задачи #25 потребовалось уменьшить объем скачиваемых данных, т.к. полученный geojson файл не получилось уместить в оперативную память. Открывал geojson файл с помощью geopandas

**Comment by GeorgiySurkov** [comment]
Существующие типы дорог в osm: https://wiki.openstreetmap.org/wiki/Key:highway

**Comment by GeorgiySurkov** [comment]
Буду перебирать возможные значения и смотреть, где они присутствуют в Санкт-Петербурге через https://overpass-turbo.eu/

**Comment by GeorgiySurkov** [comment]
По итогу выбрал подходящие значения тега.

Итоговый запрос к overpass api
```
[out:json];
area["name"="Санкт-Петербург"]->.a;
way(area.a)["highway"~"motorway|trunk|primary|secondary|tertiary|unclassified|residential|road|living_street|service|pedestrian"];
out geom;

```

**Comment by nermakovaa** [comment]
@GeorgiySurkov привет! В административную границу спб не входят некоторые части КАДа, я попробовала добавить твой запрос и столкнулась с этой проблемой, возможно стоит использовать https://nominatim.openstreetmap.org/ui/search.html для того чтобы определить границы и уже с помощью него обновить запрос. Для наглядности карту прикрепила




![Image](https://github.com/webebers/city-view/assets/114575973/ce67dadc-96d4-4b49-835b-c7992203459a)

**Comment by nermakovaa** [comment]
Кажется мы как-то так границы объединить можем (СПб и КАД), но у меня проблемы с преобразованием в geojson на этом этапе возникают, если такой запрос использовать, можешь у себя посмотреть это? @GeorgiySurkov 


[out:json];
relation(337422);
map_to_area -> .spb_area;
relation(1861646);
map_to_area -> .kad_area;
(.spb_area; .kad_area;) -> .combined_areas;
way["highway"~"motorway|trunk|primary|secondary|tertiary|unclassified|residential|road|living_street|service|pedestrian"](area.combined_areas);
out geom;

**Comment by GeorgiySurkov** [comment]
Согласен, что есть проблема с определением границ города, но не уверен, что она сейчас самая приоритетная :)

Мне кажется, что заказчику больше всего интересна топология города, а не самых его окраин. Можем уточнить этот момент, когда будем согласовывать ТЗ. 

Думаю, что мы можем сначала работать с определением границ города из QGis, потом уже это как-то поправить.

Тем более мы делаем плагин, в котором можно будет выбрать любой город, а если мы добавим конкретно такой хак для СПБ, то как тогда править границы других городов, которые также может выбрать пользователь?
